trin-core/src/portalnet/overlay.rs

```diff
@@ -175,12 +190,182 @@ impl<TContentKey: OverlayContentKey + Send, TMetric: Metric + Send>
+        if content_keys.len() != content_values.len() {
+            return Err(anyhow!(
+                "Content keys len doesn't match content values len."
+            ));
+        }
+
+        // Try to store accepted content into the database and propagate gossip the content
+        match self.protocol {
+            ProtocolId::History => {
+                // TODO: Verify history content data with Oracle
+
+                // Temporally store content key/value pairs to propagate here
+                let mut content_keys_values: Vec<(HistoryContentKey, ByteList)> = Vec::new();
+
+                for (content_key, content_value) in content_keys.into_iter().zip(content_values) {
+                    let history_key = HistoryContentKey::try_from(content_key);
+
+                    match history_key {
+                        Ok(key) => {
+                            // Store accepted content in DB
+                            self.store_accepted_content(&key, content_value.clone());
+                            content_keys_values.push((key, content_value))
+                        }
+                        Err(err) => {
+                            warn!("Unable to decode history content key: {err}");
+                            continue;
+                        }
+                    }
+                }
+                // Propagate gossip accepted content
+                self.propagate_gossip(content_keys_values);
+            }
+            ProtocolId::State => {
+                // TODO: Verify state content data with Oracle
+
+                // Temporally store content key/value pairs to propagate here
+                let mut content_keys_values: Vec<(StateContentKey, ByteList)> = Vec::new();
+
+                // Iterate over accepted content keys/values and try to store them.
+                for (content_key, content_value) in content_keys.into_iter().zip(content_values) {
+                    let history_key = StateContentKey::try_from(content_key);
+
+                    match history_key {
+                        Ok(key) => {
+                            // Store accepted content in DB
+                            self.store_accepted_content(&key, content_value.clone());
+                            content_keys_values.push((key, content_value))
+                        }
+                        Err(err) => {
+                            warn!("Unable to decode state content key: {err}");
+                            continue;
+                        }
+                    }
+                }
+                // Propagate gossip accepted content
+                self.propagate_gossip(content_keys_values);
+            }
+            _ => return Err(anyhow!("Unsupported content key protocol.")),
+        };
+        Ok(())
+    }
+
+    /// Propagate gossip accepted content via OFFER/ACCEPT:
+    fn propagate_gossip(&self, accepted_content: Vec<(impl OverlayContentKey, ByteList)>) {
```

Comment:
I think this variable name assumes too much about the caller. Wouldn't we call the same endpoint after a `portal_historyStore` for example? So maybe something like:
```suggestion
    fn propagate_gossip(&self, content: Vec<(impl OverlayContentKey, ByteList)>) {
```